### ISSUE TITLE
Keep static crates in $DDLOG_HOME.

### ISSUE THREAD

**Issue by ryzhyk** [vmware/differential-datalog#759]
Change code generation to keep static crates (d3log, cli, differential_daltalog, ovsdb) under $DDLOG_HOME instead of copying them to each generated Rust project.

**Comment by mbudiu-vmw** [comment]
Wouldn't this cause a race if multiple compilations are started concurrently?
Is there a way to lock a crate?

**Comment by ryzhyk** [comment]
I don't believe it will, as each compilation job will keep its build files locally.

**Comment by Kixiron** [comment]
That's correct, each job is separated so no race conditions will/can happen

**Comment by ryzhyk** [comment]
@convolvatron
